main.cpp

```diff
@@ -126,172 +120,108 @@ i cubed: 531441
 Use a service like https://www.diffchecker.com/diff to compare your output. 
 */
 
-#include <iostream>
-#include <cmath>
-#include <functional>
-#include <memory>
-#include <limits>
-#include <typeinfo>
-
-
 template <typename T>
 struct Numeric
 {
-    using Primitive = T;
+    using Primitive = Temporary<T>;
 
-    Numeric( Primitive varA ) : a( new Primitive(varA) ) {}
+    Numeric( Primitive varA ) : a( std::make_unique<Primitive>(varA) ) {}
 
-    operator Primitive() const { return *a; }
+    operator T() const { return *a; }
 
-    Numeric& apply( std::function< Numeric&( std::unique_ptr<Primitive>& ) > func )
-    {
-        if ( func != nullptr )
-            return func( a );
+    operator T&() { return *a; }
 
-        std::cout << "Warning, nullptr, can't apply.\n";
+    template<typename FuncPtr>
+    Numeric& apply( FuncPtr func )
+    {
+        func( a );
         return *this;
     }
 
-    using FuncPtr = void(*)( Primitive& );
-    Numeric& apply( FuncPtr func )
+    template<typename OtherType>
+    Numeric& pow( const OtherType& rhs )
     {
-        if ( func != nullptr )
-            func( *a );
+        if( a != nullptr )
+            *a = std::pow( *a, static_cast<T>(rhs) );
         return *this;
     }
 
-    Numeric& pow( Primitive rhs )
+    template<typename OtherType>
+    Numeric& operator=( const OtherType& rhs )
     {
-        return powInternal ( rhs );
+        *a = static_cast<T>(rhs);
+        return *this; 
     }
 
-    Numeric& operator+=( Primitive rhs )
+    template<typename OtherType>
+    Numeric& operator+=( const OtherType& rhs )
     {
-        *a += rhs;
+        *a += static_cast<T>(rhs);
         return *this;
     }
-
-    Numeric& operator-=( Primitive rhs )
+    
+    template<typename OtherType>
+    Numeric& operator-=( const OtherType& rhs )
     {
-        *a -= rhs;
+        *a -= static_cast<T>(rhs);
         return *this;
     }
 
-    Numeric& operator*=( Primitive rhs )
+    template<typename OtherType>
+    Numeric& operator*=( const OtherType& rhs )
     {
-        *a *= rhs;
+        *a *= static_cast<T>(rhs);
         return *this;
     }
 
-    Numeric& operator/=( Primitive rhs )
+    template<typename OtherType>
+    Numeric& operator/=( const OtherType& rhs )
     {   
         if constexpr ( std::is_same<Primitive, int>::value )
         {
-            if constexpr ( std::is_same< decltype(rhs), int>::value )
+            if constexpr ( std::is_same< OtherType, int>::value )
             { 
                 if ( rhs == 0 )
                 {
                     std::cout << "Can't divide by 0.\n";
                     return *this;
                 }
-                else if ( rhs < std::numeric_limits<Primitive>::epsilon() )
+                else if ( rhs < std::numeric_limits<OtherType>::epsilon() )
```

Comment:
`rhs` should be cast to `Primitive` and then checked against the `Primitive` epsilon.
- [ ] revise